src/protocol/libp2p/bitswap/mod.rs

```diff
@@ -287,3 +469,70 @@ impl Bitswap {
         }
     }
 }
+
```

Comment:
dq: Unrelated to these changes, we have an `self.inbound.remove(&peer)` above. However, there's a high likelyhood that if the inbound is broken, the outboud is as well. Should cleanup the outbound and force a reconnection?